src/constraints/ObjectConstrains.ts

```diff
@@ -38,18 +40,14 @@ export function whenConstraint<T extends BaseValidator<any>, I, Key extends When
 	};
 }
 
-function resolveBooleanIs<T extends BaseValidator<any>, Key extends WhenKey>(options: WhenOptions<T, Key>, value: any) {
+function resolveBooleanIs<T extends BaseValidator<any>, Key extends WhenKey>(options: WhenOptions<T, Key>, value: any, isKeyArray: boolean) {
 	if (options.is === undefined) {
-		return Boolean(value);
+		return isKeyArray ? !value.some((val: any) => !val) : Boolean(value);
 	}
 
 	if (typeof options.is === 'function') {
 		return options.is(value);
 	}
 
-	if (Array.isArray(value)) {
-		return !value.some((val) => val !== options.is);
-	}
```

Comment:
Why is this gone? This should account for `options.is` being defined `'foobar'` and checking if every value in keys `['stringLike1', 'stringLike2']` is equal to `'foobar'`

That doesn't get covered by the `.some` above which it only applies when `options.is` is `undefined`

Reply:
The check was for `Array.isArray(value)` so what'll happen is
```ts
s.object(
{ a: s.array(s.string), c: s.string.when('a', {
is: 's',
...
})
```
Here, it'll validate that the array contains only `'s'`. And I don't think we should support the array check. If we are supporting any type of object, we'll need to support all the object types (obj, map, set, ...) and that'll be out of scope for shapeshift especially when we are already providing a way (using cb) to compare using your own eq check. Also for the below case where it'll check if the array contains elements `['s']` 
```ts
s.object(
{ a: s.array(s.string), c: s.string.when('a', {
is: [ 's' ],
...
})
``` 
So then we'll need to check for deep equality of arrays too. So better just keep `===` check when `is` is provided (yup does the same ig) and if `is` isn't provided, then check 
1. if only one key is present, then `Boolean(value)` 
2. if multiple is present, `array.every(Boolean)/the one we are using`

_In my first message, I said about when `is` is undefined and multiple keys are present then check truty of each elements of the value but there was a confusion with `values` array_